### ISSUE TITLE
Refactor GraphEditor to use new mouse & layout abstractions

### ISSUE THREAD

**Issue by vilterp** [vilterp/elm-diagrams#28]
...and the broken-out modules. Would be interesting to implement panning too; see https://github.com/Ked-Ki/lx.plot

**Comment by vilterp** [comment]
blocked by #26

**Comment by vilterp** [comment]
closing in favor of a pull request: #30
